### ISSUE TITLE
Negative dice loss and IOU score more than 1

### ISSUE THREAD

**Issue by cherinae** [qubvel/segmentation_models.pytorch#415]
Hi, I followed the CamVid example and used the exact same code for the whole training process. However, the dice loss is negative, the IOU score is more than 1 and in the 100s, Fscore is also more than 1. I am unable to spot what is wrong. Please help!

```
loss = smp.utils.losses.DiceLoss()
metrics = [
    smp.utils.metrics.IoU(threshold=0.5),
    smp.utils.metrics.Fscore()
]

optimizer = torch.optim.Adam([ 
    dict(params=model.parameters(), lr=0.0001),
])

max_score = 0

for i in range(0, 30):
    
    print('\nEpoch: {}'.format(i))
    train_logs = train_epoch.run(train_loader)
    valid_logs = valid_epoch.run(valid_loader)
    
    # do something (save model, change lr, etc.)
    if max_score < valid_logs['iou_score']:
        max_score = valid_logs['iou_score']
        torch.save(model, f'/content/unet_{ENCODER}.pth')
        print('Model saved!')
        
    if i == 25:
        optimizer.param_groups[0]['lr'] = 1e-5
        print('Decrease decoder learning rate to 1e-5!')
```
![image](https://user-images.githubusercontent.com/49991539/120938887-afbdf780-c747-11eb-8cb3-dff420af364d.png)

**Comment by doursand** [comment]
I've got the same issue, i suspect that there might be a divide by batch size that is missing somewhere (at least for iou, as it converges to the batch size in my case)

**Comment by doursand** [comment]
ok in fact, my problem was that my mask was returning either 255 or 0, it needs to return 0 or 1. So just divide the mask/255.0 and it should work
